src/main/java/com/google/firebase/FirebaseOptions.java

```diff
@@ -70,10 +74,16 @@
   private final FirestoreOptions firestoreOptions;
 
   private FirebaseOptions(@NonNull FirebaseOptions.Builder builder) {
-    this.credentials = checkNotNull(builder.credentials,
-        "FirebaseOptions must be initialized with setCredentials().")
-        .createScoped(FIREBASE_SCOPES);
-    this.databaseUrl = builder.databaseUrl;
+    String emulatorUrl = EmulatorHelper.getEmulatorUrl(builder.databaseUrl);
```

Comment:
Do we need to make this check twice (once here and once in FirebaseDatabase)? It seems like FirebaseDatabase could make this check and overwrite any existing settings.

Reply:
+1 for doing this in `FirebaseDatabase`

Reply:
Done, seems cleaner.